rfcs/text/0352-graph-type-system.md

```diff
@@ -0,0 +1,2659 @@
+Property Types will be defined similarly to Entity Types in the proposed system, conforming to the [Property Type](#property-types-1) schemas instead.
+Link Types do not contain further structural data, which make them semantically simpler than Entity Types and Property Types.
+
+**Entity Type Create and Update messages in the current system:**
+
+```json
+{
+  // createEntityType message
+  "schema": {
+    "type": "object",
+    "name": "Person",
+    "properties": {
+      "name": { "type": "string" },
+      "email": { "type": "string" },
+      "phoneNumber": { "type": "string" }
+    }
+  }
+}
+```
+
+```json
+{
+  // updateEntityType message
+  "entityTypeId": "Person",
+  "schema": {
+    "properties": {
+      "birthDate": { "type": "string", "format": "date" }
+    }
+  }
+}
+```
+
+**Entity Type Create and Update messages in the proposed system:**
+
+```json
+{
+  // createEntityType message
+  "schema": {
+    "type": "object",
+    "kind": "entityType",
+    "name": "Person",
+    "properties": {
+      "https://blockprotocol.org/types/@alice/property-type/name": {
+        "$ref": "https://blockprotocol.org/types/@alice/property-type/name"
+      },
+      "https://blockprotocol.org/types/@alice/property-type/email": {
+        "$ref": "https://blockprotocol.org/types/@alice/property-type/email"
+      },
+      "https://blockprotocol.org/types/@alice/property-type/phone-number": {
+        "$ref": "https://blockprotocol.org/types/@alice/property-type/phone-number"
+      }
+    }
+  }
+}
+```
+
+```json
+{
+  // updateEntityType message
+  "entityTypeId": "Person",
+  "schema": {
+    // The properties here are partially applied to the original Entity Type.
+    "properties": {
+      "https://blockprotocol.org/types/@alice/property-type/birth-date": {
+        "$ref": "https://blockprotocol.org/types/@alice/property-type/birth-date"
+      }
+    }
+  }
+}
+```
+
+The update messages of both the current and new systems make use of partial schemas, merging the schema given in the message contents with the existing Entity Type. This may or may not be the desired semantics of updating, and could lead to undesired behavior. In that case, the semantics can be changed to treat updates as a complete replacement.
+
+> 💡 The `entityTypeId` here refers to the unique name of the Entity Type. It could make use of the Entity Type's canonical URI instead.
+
+**Property Type Create and Update messages in the current system:**
+
+Property Types do not exist.
+
+**Property Type Create and Update messages in the proposed system:**
+
+```json
+{
+  // createPropertyType message
+  "schema": {
+    "kind": "propertyType",
+    "name": "User ID",
+    "oneOf": [
+      {
+        "$ref": "https://blockprotocol.org/types/@blockprotocol/data-type/text"
+      }
+    ]
+  }
+}
+```
+
+```json
+{
+  // updatePropertyType message
+  "propertyTypeId": "User ID",
```

Comment:
`name` is not a unique identifier, surely this should be whatever its `$id` is?

Reply:
Addressed in https://github.com/blockprotocol/blockprotocol/pull/352/commits/644bfc128e2b0c55e907edff429e6dfa3843dab3, fixing other similar places